virtual_rainforest/models/abiotic/energy_balance.py

```diff
@@ -0,0 +1,321 @@
+        soil_node_depths: NDArray[np.float32],
+        height_of_above_canopy: NDArray[np.float32],
+        relative_humidity_2m: NDArray[np.float32],
+        air_temperature_2m: NDArray[np.float32],
+        mean_annual_temperature: NDArray[np.float32], temperature of deepest soil layer
+        atmospheric_pressure_2m: NDArray[np.float32],
+        absorbed_radiation: : NDArray[np.float32],
+        air_conductivity: NDArray[np.float32],
+        leaf_conductivity: NDArray[np.float32],
+        air_leaf_conductivity: NDArray[np.float32],
+        sensible_heat_flux: NDArray[np.float32],
+        latent_heat_flux: NDArray[np.float32],
+        ground_heat_flux: NDArray[np.float32],
+        diabatic_correction_factor: NDArray[np.float32]
+        soil_type: NDArray[np.string_]
+        soil_depth: NDArray[np.float32]
+        soil_moisture: NDArray[np.float32]
+        soil_moisture_reference: NDArray[np.float32]
+        wind_profile: NDArray[np.float32]
+    """
+
+    def __init__(
+        self,
+        data: dict[str, float],
+        soil_type: NDArray[np.string_],
+        soil_depth: NDArray[np.float32] = np.array(2.0, dtype=float),
+        mean_annual_temperature: NDArray[np.float32] = np.array(20, dtype=float),
+        canopy_layers: NDArray[np.int32] = np.array(3, dtype=int),
+        soil_layers: NDArray[np.int32] = np.array(2, dtype=int),
+        canopy_height: NDArray[np.float32] = np.array(30.0, dtype=float),
+        absorbed_radiation: NDArray[np.float32] = np.array(100.0, dtype=float),
+    ) -> None:
+        """Initializes point-based energy_balance method.
+
+        Generates a set of climate, conductivity, and soil parameters for running the
+        first time step of the model. This might go in the json file.
+        """
+
+        # set boundary conditions
+        self.canopy_layers = canopy_layers
+        self.soil_layers = soil_layers
+        self.canopy_height = canopy_height
+        self.soil_depth = soil_depth
+
+        self.air_temperature_2m = data["air_temperature_2m"]
+        self.relative_humidity_2m = data["relative_humidity_2m"]
+        self.atmospheric_pressure_2m = data["atmospheric_pressure_2m"]
+        self.wind_speed_10m = data["wind_speed_10m"]
+        self.mean_annual_temperature = mean_annual_temperature
+        self.soil_moisture_reference = data["soil_moisture_reference"]
+
+        # interpolate initial temperature profile, could be more realistic
+        self.temperature_above_canopy = self.air_temperature_2m
+        temperature_interpolation = np.linspace(
+            self.mean_annual_temperature,
+            self.temperature_above_canopy,
+            self.canopy_layers + self.soil_layers,
+        )
+
+        self.air_temperature = temperature_interpolation[
+            (int(self.soil_layers)) : int((self.canopy_layers + self.soil_layers))
+        ]
+        self.soil_temperature = temperature_interpolation[0 : int(self.soil_layers)][
+            ::-1
+        ]
```

Comment:
What's the `[::-1]` on the end for? Is it because you want to cast to a `np.array` or something?

Reply:
As I think I mentioned before, `self.soil_layers` should probably just be of type `int`. There's no point its being a `np.array` if it's a scalar.